python/main.py

```diff
@@ -94,12 +110,39 @@ async def get_image(image_name):
 
     return FileResponse(image)
 
+async def hash_and_rename_image(image: UploadFile):
+    image_binary = await image.read()
+    
+    # SHA-256 ハッシュを計算
+    hash_value = hashlib.sha256(image_binary).hexdigest()
+    
+    # 新しいファイル名を生成
+    image_name = f"{hash_value}.jpg"
+    
+    # ファイルをリネーム
+    os.rename(image_binary, image_name)
```

Comment:
```suggestion
       # Save the image with the new name
    image_path = images / image_name
    with open(image_path, "wb") as f:
        f.write(image_binary)
```
<p>
The process here shouldn't be renaming the existing image file. It's rather more about adding a new image file for this image. It's like uploading an image to somewhere online. We create a new image that store somewhere instead of renaming it. 
For example, if multiple items has the same image upload, we will still store them in separate file because they belong to different items. If we store them as one file, then if someone delete it, all other items will be affected.
</p>
-----
<p>
こちらのプロセスは既存している画像をRenameするより、既存している画像からデータを取ってHashかしてアップロードすることに成るはずです。
実際の例に言うと、Websiteに画像をアップロードする時、Localのファイルがアップロードされているんですが、そのファイル自体のNameや他の情報が変わりません。Step４は全部Localでやっているため多少わかりにくいかもしれないですが、実際のサービスだったらとあるDatabaseの中に入れることになります。そこで、たとえ複数のItemが同じImageを共有していても、別々で保存することになります
</p>